### PR TITLE
Checkout: Improve post-renewal copy and fix "learn more" link

### LINKED ISSUE #324: Editor: Allow dropping on featured image
Suggested by @mtias:

> we should set a drag-and-drop area on the featured image upload placeholder.

We should enhance the Featured Image editor sidebar by making it a droppable area (using the <DropZone /> component).

screen shot 2015-11-20 at 3 27 50 pm

### PR DESCRIPTION
**Pull request by chrisfromthelc**
#### Changes proposed in this Pull Request

This pull request addresses post-renewal notice copy for renewals that are set to auto-renew going forward. It replaces the text `{Plan} has been renewed and will now auto renew in the future. Learn more.` ([old link target](https://wordpress.com/support/manage-purchases/)) pointing to the main Manage Purchases support doc, with `'Success! You renewed {productName} for {duration}, and we sent your receipt to {email}. Learn more about renewals.` ([new link target](https://wordpress.com/support/manage-purchases/#automatic-renewal)) pointing to the specific renewals section in the Manage Purchases support doc.

Old version:
Screen Shot 2021-07-28 at 3 39 45 PM

New version:
Screen Shot 2021-07-28 at 3 38 07 PM

For plans that cannot autorenew due to the payment type (WordPress.com credits, non-US transactions, etc.), they will be shown the text without the autorenewal link: `'Success! You renewed {productName} for {duration}, and we sent your receipt to {email}.`.

#### Testing instructions

Renew any plan using a test credit card number.

The message should display as below, along with the link that should lead to https://wordpress.com/support/manage-purchases/#automatic-renewal.

Screen Shot 2021-07-28 at 3 38 07 PM

Fixes #324

### CODE REVIEW

**Comment by sirbrillig** [comment]
I updated the title of this PR to `Checkout: Improve post-renewal copy and fix "learn more" link` because I felt that `Fix/post checkout copy and fix learn more link` wasn't very clear. Feel free to change it again if you come up with something better!

I'd also suggest maybe including information in the PR description about what was fixed about the "learn more" link. Currently it explains that the text is being changed, but it doesn't mention anything about the link (apart from a reference to "link target" without much context). It could also be that nothing was broken about the "learn more" link after all and I mistakenly kept that wording in the title, in which case we can remove it.

File: client/my-sites/checkout/composite-checkout/hooks/use-create-payment-complete-callback.tsx [hunk 1]
```diff
@@ -276,11 +276,14 @@ function displayRenewalSuccessNotice(
...
 				translate(
-					'%(productName)s has been renewed and will now auto renew in the future. ' +
-						'{{a}}Learn more{{/a}}',
+					'Success! You renewed %(productName)s for %(duration)s, and we sent your receipt to %(email)s. ' +
+						'{{a}}Learn more about renewals{{/a}}',
 					{
 						args: {
 							productName: renewalItem.product_name,
+							duration: moment.duration( { days: renewalItem.bill_period } ).humanize(),
+							renewalDate: moment( product.expiry ).subtract( 31, 'days' ).format( 'LL' ),
```
Comment (sirbrillig) [inline client/my-sites/checkout/composite-checkout/hooks/use-create-payment-complete-callback.tsx]: I could be missing something, but I don't think that `renewalDate` is used anywhere in this string?

File: client/my-sites/checkout/composite-checkout/hooks/use-create-payment-complete-callback.tsx [hunk 2]
```diff
@@ -276,11 +276,14 @@ function displayRenewalSuccessNotice(
 		reduxDispatch(
 			successNotice(
 				translate(
-					'%(productName)s has been renewed and will now auto renew in the future. ' +
-						'{{a}}Learn more{{/a}}',
+					'Success! You renewed %(productName)s for %(duration)s, and we sent your receipt to %(email)s. ' +
```
Comment (sirbrillig) [inline client/my-sites/checkout/composite-checkout/hooks/use-create-payment-complete-callback.tsx]: Minor, and I know it was this way before, but I don't see any need to split this string into two lines and join them. I think it decreases readability by making it seem that there's two separate strings being translated here or that there's a newline included. That may be a subjective opinion, though, so feel free to ignore it if you disagree.

**Review by sirbrillig — COMMENTED** [review]
Tested and it works as described! 

success-message


I left one comment which I think needs to be taken care of but otherwise 👍

File: client/my-sites/checkout/composite-checkout/hooks/use-create-payment-complete-callback.tsx (hunk 1 above)
Comment (chrisfromthelc) [reply client/my-sites/checkout/composite-checkout/hooks/use-create-payment-complete-callback.tsx]: Sorry, I'd thought I'd removed this already; this was from a previous iteration of the message.

File: client/my-sites/checkout/composite-checkout/hooks/use-create-payment-complete-callback.tsx (hunk 2 above)
Comment (chrisfromthelc) [reply client/my-sites/checkout/composite-checkout/hooks/use-create-payment-complete-callback.tsx]: I wasn't sure if that was intentional, but I agree. I've made it all one string with no apparent ill effects, and it also appears to work fine across multiple browser widths.
Screen Shot 2021-07-29 at 8 59 32 AM

Screen Shot 2021-07-29 at 9 00 58 AM

I've done the same for the non-auto-renew notification as well, as it was also split into multiple strings.
Screen Shot 2021-07-29 at 9 03 46 AM

**Review by sirbrillig — APPROVED** [review]
Thanks for making those changes. The diff looks good to me now.

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/6301359

Thank you @chrisfromthelc for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/6301359).

**Pull request merged**